File: tmdb-core/src/main/java/dev/reuss/tmdb/domain/tv/series/model/TvSeriesSimilar.java
package dev.reuss.tmdb.domain.tv.series.model;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;
import dev.reuss.tmdb.common.TmdbModel;
import dev.reuss.tmdb.common.page.PagedResponse;
import dev.reuss.tmdb.util.TmdbCollections;

import java.util.List;

/**
 * Similar TV series for a TMDB TV series.
 *
 * @param page         current page
 * @param results      similar TV series
 * @param totalPages   total pages
 * @param totalResults total results
 */
@JsonIgnoreProperties(ignoreUnknown = true)
public record TvSeriesSimilar(
        int page,

        List<TvSeriesSimilarItem> results,

        @JsonProperty("total_pages")
        int totalPages,

        @JsonProperty("total_results")
        int totalResults
) implements PagedResponse<TvSeriesSimilarItem>, TmdbModel {

    public TvSeriesSimilar {
        results = TmdbCollections.list(results);
    }
}

File: tmdb-core/src/main/kotlin/dev/reuss/tmdb/domain/tv/series/model/TvSeriesSimilar.kt
package dev.reuss.tmdb.domain.tv.series.model

import com.fasterxml.jackson.annotation.JsonProperty
import dev.reuss.tmdb.common.TmdbModel
import dev.reuss.tmdb.common.page.PagedResponse

/**
 * Similar TV series for a TMDB TV series.
 *
 * @property page current page
 * @property results similar TV series
 * @property totalPages total pages
 * @property totalResults total results
 */
@JvmRecord
data class TvSeriesSimilar(
    @all:JsonProperty("page")
    override val page: Int,

    @all:JsonProperty("results")
    override val results: List<TvSeriesSimilarItem> = emptyList(),

    @all:JsonProperty("total_pages")
    override val totalPages: Int,

    @all:JsonProperty("total_results")
    override val totalResults: Int
) : PagedResponse<TvSeriesSimilarItem>, TmdbModel